### PR TITLE
Fixed: Accessibility: Time picker unreachable with keyboard navigation #2144

### PR DESCRIPTION
**Pull request by shubhamjajoo**
Added **onkeyDown** event handler to handle the key code events and based on the buttons pressed  operating over date list.
Now user can select the date upon pressing either _Space_ /_Enter_.

### CODE REVIEW

File: src/time.jsx [hunk 1]
```diff
@@ -68,6 +76,45 @@ export default class Time extends React.Component {
     }
   }
 
+  onKeyDown = event => {
+    if (event.keyCode !== KEY_CODE.TAB) {
+      event.preventDefault();
+      const activeEle = document.activeElement;
+      if (event.keyCode === KEY_CODE.DOWN) {
+        const nextEle = document.activeElement.nextElementSibling;
```
Comment (jwallet) [inline src/time.jsx:84]: not sure about using dom query, we have acess to react and ref elements
I will prefer to see this implemented in index.js where the accessibility for the calendar is implemented

Comment (shubhamjajoo) [reply src/time.jsx]: @jwallet, Handled accessibility logic in index.js.  
let me know if this is working as expected.

**Comment by martijnrusschen** [comment]
Can you add a test to make sure this functionality keeps working in the future?

**Comment by devangpatel21** [comment]
can you please merge it or add some another way for this @jwallet

**Comment by devangpatel21** [comment]
can you fixed and upload it back @shubhamjajoo ?

**Comment by shubhamjajoo** [comment]
> can you fixed and upload it back @shubhamjajoo ?

**Comment by shubhamjajoo** [comment]
Updated @devangpatel21 , seems   **✖ should update the preSelection state when a day is selected with mouse click**
test case is failing

**Comment by devangpatel21** [comment]
So is there any different way for this problem bcoz I need the accessibility for time picker @shubhamjajoo @jwallet

**Comment by devangpatel21** [comment]
> Updated @devangpatel21 , seems **✖ should update the preSelection state when a day is selected with mouse click**
> test case is failing

can something be done for this @jwallet?

**Comment by devangpatel21** [comment]
any update @shubhamjajoo

**Comment by devangpatel21** [comment]
> What is the status of this? Would really like to have this!!!
> 
> @shubhamjajoo

any updates?

**Comment by devangpatel21** [comment]
@jwallet can you suggest any solution for this issue

**Comment by martijnrusschen** [comment]
Closing due to inactivity, please reopen if you want to finish this.

**Pull request closed without merging**